spec/components/jobseekers/schools_overview_component_spec.rb

```diff
@@ -99,14 +106,63 @@
   context 'when GIAS-obtained website has been overwritten' do
     it 'renders a link to the school website' do
       expect(rendered_component).to include(school_1.website)
+      expect(rendered_component).not_to include(school_1.url)
     end
   end
 
-  it 'renders all the school links to the school website' do
+  it 'renders all the GIAS-provided links to the school website' do
     [school_2, school_3].each { |school| expect(rendered_component).to include(school.url) }
   end
 
   it 'renders the school visits' do
     expect(rendered_component).to include(vacancy.school_visits)
   end
+
+  it 'renders the location heading for multiple schools' do
+    expect(rendered_component).to include('School locations')
+  end
+
+  context 'when at least one school has a geolocation' do
+    it 'shows the map' do
+      expect(rendered_component).to include('School locations')
+      expect(rendered_component).to include('map')
+    end
+  end
+
+  context 'when no school has a geolocation' do
+    let(:geolocation_trait) { :no_geolocation }
```

Comment:
I'd prefer us to stub `any?(&:geolocation)` and remove the need for `geolocation_trait`